### PR TITLE
doc: matter: persistent subscriptions and Matter docset version

### PR DESCRIPTION
**Pull request by greg-fer**
Extended advanced kconfigs docs to include information about
Matter persistent subscriptions configuration.
Added Matter version to the docset.

----

This PR has been reworked in scope due to the delay of the Matter dev tag. It only includes updates not related to the dev tag.

### CODE REVIEW

File: doc/nrf/protocols/matter/index.rst [hunk 1]
```diff
@@ -22,6 +22,8 @@ The |NCS| allows you to develop applications with different versions of Matter,
 +--------------------------+-----------------------------------------------------+------------------------+
 | nRF Connect SDK version  | Matter specification version                        | `Matter SDK version`_  |
 +==========================+=====================================================+========================+
+| v2.3.0-dev1              | :ref:`1.1.0 <ug_matter_overview_dev_model_support>` | 1.1.0.0                |
```
Comment (umapraseeda) [inline doc/nrf/protocols/matter/index.rst:25]:
```suggestion
| v2.3.99-dev1             | :ref:`1.1.0 <ug_matter_overview_dev_model_support>` | 1.1.0.0                |
```

File: doc/versions.json [hunk 2]
```diff
@@ -44,7 +45,16 @@
       "mcuboot": "1.9.99",
       "nrfx": "2.10",
       "tfm": "1.6.0",
-      "matter": ""
+      "matter": "1.1.0"
```
Comment (umapraseeda) [inline doc/versions.json:48]: fyi: https://github.com/nrfconnect/sdk-nrf/pull/10683/files

File: doc/nrf/index.rst [hunk 3]
```diff
@@ -3,6 +3,18 @@
 Welcome to the |NCS|!
 #####################
 
+.. important::
+
+   |NCS| v2.3.99-dev1 is a development tag and will be replaced by v2.4.0 in the future.
+
+   The development tag contains the following major updates:
```
Comment (umapraseeda) [inline doc/nrf/index.rst:10]: Please ignore if you have more updates to list.
```suggestion
   The development tag contains the following major update:
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -139,25 +139,15 @@ This section provides detailed lists of changes by :ref:`application <applicatio
 nRF9160: Asset Tracker v2
 -------------------------
 
-* Updated:
-
-  * Use defines from the :ref:`lib_nrf_cloud` library for nRF Cloud related string values.
-  * The application now integrates the :ref:`lib_lwm2m_client_utils` FOTA callback functionality.
+* Added the integration of the :ref:`lib_lwm2m_client_utils` FOTA callback functionality.
+* Updated use defines from the :ref:`lib_nrf_cloud` library for nRF Cloud-related string values.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:143]:
```suggestion
* Updated the application to use defines from the :ref:`lib_nrf_cloud` library for nRF Cloud-related string values.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -139,25 +139,15 @@ This section provides detailed lists of changes by :ref:`application <applicatio
...
-
-* Updated:
-
-  * Use defines from the :ref:`lib_nrf_cloud` library for nRF Cloud related string values.
-
-* Fixed:
-
-  * A bug in receiving large MQTT Publish message.
+* Added AT command ``#XWIFIPOS`` to get Wi-Fi location from nRF Cloud.
+* Updated use defines from the :ref:`lib_nrf_cloud` library for nRF Cloud related string values.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:149]:
```suggestion
* Updated the application to use defines from the :ref:`lib_nrf_cloud` library for nRF Cloud related string values.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 6]
```diff
@@ -172,7 +162,8 @@ nRF Machine Learning (Edge Impulse)
 nRF Desktop
 -----------
 
-* Added the :ref:`nrf_desktop_swift_pair_app`. The module is used to enable or disable the Swift Pair Bluetooth advertising payload depending on the selected Bluetooth peer (used local identity).
+* Added the :ref:`nrf_desktop_swift_pair_app`.
+  The module is used to enable or disable the Swift Pair Bluetooth advertising payload depending on the selected Bluetooth peer (used local identity).
 
 * Updated:
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:169]: FLASH -> flash in line 179,180

File: doc/nrf/releases/release-notes-changelog.rst [hunk 7]
```diff
@@ -202,28 +191,20 @@ nRF9160 samples
 
 * :ref:`modem_shell_application` sample:
 
-  * Updated:
-
-    * Use defines from the :ref:`lib_nrf_cloud` library for nRF Cloud related string values.
-      Remove the inclusion of the file :file:`nrf_cloud_codec.h`.
+  * Updated use defines from the :ref:`lib_nrf_cloud` library for nRF Cloud related string values.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:194]:
```suggestion
  * Updated the sample to use defines from the :ref:`lib_nrf_cloud` library for nRF Cloud related string values.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 8]
```diff
@@ -202,28 +191,20 @@ nRF9160 samples
...
-
-    * The sample now integrates the :ref:`lib_lwm2m_client_utils` FOTA callback functionality.
+  * Updated the sample to integrate the :ref:`lib_lwm2m_client_utils` FOTA callback functionality.
 
 * :ref:`nrf_cloud_mqtt_multi_service` sample:
 
-  * Updated:
-
-    * Increased the MCUboot partition size to the minimum necessary to allow bootloader FOTA.
+  * Updated the sample by increasing the MCUboot partition size to the minimum necessary to allow bootloader FOTA.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:207]: Line 87: :ref:\`matter_lock_sample\` ->:ref:\`matter_lock_sample\` sample
Line 246: * :ref:\`matter_lock_sample\` sample:
Line 305: * :ref:\`bt_fast_pair_readme\` library:

**Comment by divipillai** [comment]
@greg-fer : Could you please add this https://github.com/nrfconnect/sdk-nrf/pull/10665#discussion_r1146059230 suggestion to this PR. This could not be fixed in the original PR.

File: doc/nrf/protocols/matter/overview/dev_model.rst [hunk 9]
```diff
@@ -39,6 +39,8 @@ The following table lists Matter versions supported in the |NCS|, with a brief o
 +-----------------+----------------------------------------------+---------------------+
 | Matter version  | Overview of changes                          | Release date        |
 +=================+==============================================+=====================+
+| 1.1.0           |                                              |                     |
```
Comment (kkasperczyk-no) [inline doc/nrf/protocols/matter/overview/dev_model.rst:42]: As we discussed F2F, here is a list of changes:

- Introduced support for persistent subscriptions
- Added certification tests verifying usage of multiple credentials types at the same time, such as pin codes, RFID or fingerprint, for a Door Lock device type (cluster implementation was present in Matter 1.0, but it was not verified).
- Introduced significant editorial changes and clarifications in the specification, improved alignment of the Matter specification with the Matter SDK , and added several bug fixes.

**Review by kkasperczyk-no — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**